.github/workflows/ci.yml

```diff
@@ -40,26 +40,76 @@ jobs:
     strategy:
       matrix:
         os:
-          - 'amazon-linux'
-          - 'debian-9'
-          - 'debian-10'
-          - 'centos-7'
-          - 'ubuntu-1604'
-          - 'ubuntu-1804'
+          - amazon-linux
+          - amazonlinux-2
+          - debian-9
+          - debian-10
+          - centos-7
+          - ubuntu-1604
+          - ubuntu-1804
+          - ubuntu-2004
         suite:
-          - 'openjdk-11'
-          - 'openjdk-12'
-          - 'openjdk-13'
-          - 'adoptopenjdk-8-openj9'
-          - 'adoptopenjdk-11-hotspot'
-          - 'adoptopenjdk-11-openj9'
-          - 'adoptopenjdk-11-openj9-large-heap'
-          - 'adoptopenjdk-13-hotspot'
-          - 'adoptopenjdk-13-openj9'
-          - 'adoptopenjdk-13-openj9-large-heap'
-          - 'adoptopenjdk-removal-11-openj9'
-          - 'corretto-11'
-          - 'corretto-8'
+          - openjdk-11
+          - openjdk-12
+          - openjdk-13
+          - openjdk-pkg-7
+          - openjdk-pkg-8
+          - openjdk-pkg-11
+          - openjdk-pkg-13
+          - openjdk-pkg-latest
+          - adoptopenjdk-8-openj9
+          - adoptopenjdk-11-hotspot
+          - adoptopenjdk-11-openj9
+          - adoptopenjdk-11-openj9-large-heap
+          - adoptopenjdk-13-hotspot
+          - adoptopenjdk-13-openj9
+          - adoptopenjdk-13-openj9-large-heap
+          - adoptopenjdk-removal-11-openj9
+          - corretto-11
+          - corretto-8
+        exclude:
+          - os: amazon-linux
+            suite: openjdk-pkg-11
+          - os: amazon-linux
+            suite: openjdk-pkg-13
+          - os: amazon-linux
+            suite: openjdk-pkg-latest
+          - os: amazonlinux-2
+            suite: openjdk-pkg-13
+          - os: amazonlinux-2
+            suite: openjdk-pkg-latest
+          - os: centos-7
+            suite: openjdk-pkg-13
+          - os: centos-7
+            suite: openjdk-pkg-latest
+          - os: debian-9
+            suite: openjdk-pkg-7
+          - os: debian-9
+            suite: openjdk-pkg-11
+          - os: debian-9
+            suite: openjdk-pkg-13
+          - os: debian-9
+            suite: openjdk-pkg-latest
+          - os: debian-10
+            suite: openjdk-pkg-7
+          - os: debian-10
+            suite: openjdk-pkg-8
+          - os: debian-10
+            suite: openjdk-pkg-13
+          - os: debian-10
+            suite: openjdk-pkg-latest
+          - os: ubuntu-1604
+            suite: openjdk-pkg-latest
+          - os: ubuntu-1804
+            suite: openjdk-pkg-7
+          - os: ubuntu-1804
+            suite: openjdk-pkg-13
+          - os: ubuntu-1804
+            suite: openjdk-pkg-latest
+          - os: ubuntu-2004
+            suite: openjdk-pkg-7
+          - os: ubuntu-2004
+            suite: openjdk-pkg-latest
```

Comment:
I might reduce this list, as I think we're going to get into unmanageable amounts of integration tests! 😆

Reply:
yeah I wanted to have testing coverage but really started to struggle with amazon, rhel(family), debian, ubuntu, all having varying versions in their repos and different versions of a specific OS having older java versions but not newer or vice versa. Thus the various excludes. 

Oracle has really fractured the java landscape.